### PR TITLE
Prevent swallowing of error during debian creation with the final close

### PR DESCRIPTION
**Pull request by chali**
We have just hit the same problem as #234 What do you think about following approach for those situations?

### CODE REVIEW

File: src/test/java/org/vafer/jdeb/DebMakerTestCase.java [hunk 1]
```diff
@@ -21,12 +21,7 @@
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (tcurdt) [inline src/test/java/org/vafer/jdeb/DebMakerTestCase.java:24]: Please no ".*" imports.

File: src/main/java/org/vafer/jdeb/DataBuilder.java [hunk 2]
```diff
@@ -272,12 +273,20 @@ private void createParentDirectories( String filename, String user, long uid, St
...
         };
 
+        boolean finishedWithoutErrors = true;
         try {
             for (DataProducer data : producers) {
                 data.produce(receiver);
             }
+        } catch(Exception e) {
+            finishedWithoutErrors = false;
+            throw new RuntimeException("Couldn't process all data producers", e);
```
Comment (tcurdt) [inline src/main/java/org/vafer/jdeb/DataBuilder.java:283]: you want the RuntimeException so that fails the build IIUC?

**Comment by tcurdt** [comment]
Thanks for the PR!

I guess the question I have is:

Why wouldn't an exception in the producer bubble up to produce the very same error condition?

File: src/main/java/org/vafer/jdeb/DataBuilder.java (hunk 2 above)
Comment (chali) [reply src/main/java/org/vafer/jdeb/DataBuilder.java:283]: I was thinking about using using `PackagingException` but that would require adding to method signature. Could that be potential breaking change?

File: src/test/java/org/vafer/jdeb/DebMakerTestCase.java (hunk 1 above)
Comment (chali) [reply src/test/java/org/vafer/jdeb/DebMakerTestCase.java:24]: will fix that.

**Comment by chali** [comment]
The situation that is happening is if there is an `Exception` in between `tarOutputStream.putArchiveEntry(fileEntry);` and `tarOutputStream.closeArchiveEntry();` like full temp space in the linked issue. It leaves tar entry in unclosed state. Then calling `tarOutputStream.close()` that is in the `finally` causes another exception. The first one is ignored and the second one is propagated outside. Leaving users without clue what actually happened.

File: src/main/java/org/vafer/jdeb/DataBuilder.java (hunk 2 above)
Comment (chali) [reply src/main/java/org/vafer/jdeb/DataBuilder.java:283]: The most interesting context is the caught exception.

File: src/main/java/org/vafer/jdeb/DebMaker.java [hunk 3]
```diff
@@ -22,7 +22,6 @@
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang3.StringUtils;
```
Comment (chali) [inline src/main/java/org/vafer/jdeb/DebMaker.java:25]: Unused import

File: src/test/java/org/vafer/jdeb/DebMakerTestCase.java [hunk 4]
```diff
@@ -25,14 +25,14 @@
...
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 import org.apache.commons.compress.archivers.ar.ArArchiveEntry;
 import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.output.NullOutputStream;
-import org.apache.commons.lang3.ArrayUtils;
```
Comment (chali) [inline src/test/java/org/vafer/jdeb/DebMakerTestCase.java:35]: Unused import

File: src/main/java/org/vafer/jdeb/DataBuilder.java (hunk 2 above)
Comment (chali) [reply src/main/java/org/vafer/jdeb/DataBuilder.java]: ah, learning a bit how exception handling :-D  changed to just rethrow original exception.

**Comment by tcurdt** [comment]
Looking better! Thanks for the contribution!

Feel free to also include yourself as contributor https://github.com/tcurdt/jdeb/blob/master/pom.xml#L88
And this also still needs to be added to https://github.com/tcurdt/jdeb/blob/master/HISTORY.md

...and I just noticed a new release is overdue 🙃

**Comment by chali** [comment]
> Feel free to also include yourself as contributor https://github.com/tcurdt/jdeb/blob/master/pom.xml#L88
> And this also still needs to be added to https://github.com/tcurdt/jdeb/blob/master/HISTORY.md

Working on those. https://github.com/tcurdt/jdeb/pull/708

> ...and I just noticed a new release is overdue 🙃

:-D little bit

**Comment by rpalcolea** [comment]
hey @tcurdt , thanks for looking at this contribution! 

Wondering if you have any thoughts on a potential release soon? :)

**Comment by tcurdt** [comment]
Yeah - about that... :)
Thanks for the reminder!

I hope the maven central config still works.
There were some changes recently.

I'll try to get it done.

**Comment by tcurdt** [comment]
Done! https://github.com/tcurdt/jdeb/releases/tag/jdeb-1.11

**Comment by tcurdt** [comment]
Still waiting for maven central to sync

**Comment by rpalcolea** [comment]
awesome! thank you very much!

**Pull request merged**